### PR TITLE
Adds `backwards` property and `play_backwards` method to `AnimatedSprite`

### PR DESCRIPTION
**Pull request by robfram**
Adds `backwards` property and `play_backwards` method to `AnimatedSprite`

This PR adds the posibility to play `AnimationSprite` in straigh or
reverse direction, using a new property and adding a new optional
argument to the `play` method (default value has the same old
behaviour). A new method (`play_backwards`) is added for convenience,
but its kind of an alias to the new argument of `play`.

Also fixes a typo in `is_playing` method description (Documentation).

It should implement #17787.

### CODE REVIEW

**Comment by Vega-KH** [comment]
It looks like I was working on AnimatedSprite at the same time as you. My PR will probably conflict with this somewhere.

BTW, I was thinking about adding the same thing you did, but my idea for implementation was to play the animation backwards if the animation speed was negative.

**Comment by robfram** [comment]
@Vega-KH Sorry! I did this yesterday as it was more than ten days old and didn't see any post :(

About your implentation, it was my first idea, as `AnimationPlayer` does that way with its `speed` parameter. But later I realized that perhaps FPS (as in `AnimatedSprite` is what `speed` means actually) has no sense being negative. You will change to the next frame or the previous frame, but the number of frames changed per second will remain the same. There is no `speed` parameter in the `play` method, and normally the frame per frame animation is stick with a specific speed to be visually pleasant. Code uses `time` to compute remaining time to change frames emiting singals, and perhaps having it negative will result in a bit more complex code. Also, code is using `time > 0`  as error checking, so perhaps it is more delicate to change. In the end, I thought it was more intuitive to have a _digital lever_ to choose the direction of the animation. 

TL;DR: I thought it was more intuitive to add a _digital lever_ to choose the direction of the animation and don't mess with negative FPS values in the `speed` property.

It is only my perception. The bosses should choose which one seems more _godotic_ :)

**Comment by reduz** [comment]
IMO this should be an argument to play() or a property, but not both. I would add as an argument to play() only, for consistency with AnimationPlayer

**Comment by akien-mga** [comment]
Closing as stale, but a new PR could be opened to implement this in the way suggested by @reduz.
